tests/basetest.rb

```diff
@@ -100,7 +100,40 @@ def teardown
   end
 
   def config(*args)
-    @device.cmd("configure terminal\n" + args.join("\n") + "\nend")
+    # Send the entire config as one string but be sure not to return until
+    # we are safely back out of config mode, i.e. prompt is
+    # 'switch#' not 'switch(config)#' or 'switch(config-if)#' etc.
+    @device.cmd('String' => "configure terminal\n" + args.join("\n") + "\nend",
+                'Prompt' => /^[^()]+[$%#>] \z/n)
```

Comment:
`cmd()` doesn't return until it sees a string that matches the given Prompt regex. This regex matches a line that has a series of any characters that do NOT include '(' or ')', ending in one of the standard prompt characters `$%#>` followed by a single space.

The exclusion of '(' and ')' means that we will not match config-mode prompt lines like `switch(config)# ` or `n9k-108(config-if)# ` - we will only match (and be correctly 'done' with the config command) when we get an exec-mode prompt like `switch# ` or `uut.cisco.com# `.

I was hoping the comment above made this clear enough  but I guess not. What can I add to make this more self-evident?